lib/kube/proxy/resource_filters.go

```diff
@@ -173,10 +178,10 @@ func (d *resourceFilterer) FilterObj(obj runtime.Object) (isAllowed bool, isList
 	case *corev1.Pod:
 		result, err := filterResource(d.kind, d.verb, o, d.allowedResources, d.deniedResources)
 		if err != nil {
-			d.log.WithError(err).Warn("Unable to compile role kubernetes_resources.")
+			d.log.WithError(err).Warn("Unable to compile regex expressions within kubernetes_resources.")
 		}
 		// if err is not nil or result is false, we should not include it.
-		return err == nil && result, false, nil
+		return result, false, nil
```

Comment:
I'd still say that this change introduces dependency on something that is not exactly a strong contract of the `filterResource` function and is neither enforced by the compiler, nor spelled out directly in the function's documentation. The previous code adhered to the principle of being permissive about your inputs (here: the values returned from `filterResource()`) and strict about your outputs (here: the value returned by `FilterObj()`).

Reply:
The previous code only returned an error and didn't support the `result`.  Due to performance issues associated with `trace.Wrap`, we included the result but we never change these functions